### PR TITLE
65748

### PR DESCRIPTION
**Pull request by jakaria-istauk**

### CODE REVIEW

File: includes/Elements/Data_Table.php [hunk 1]
```diff
@@ -1400,7 +1395,7 @@ protected function render( ) {
 										'style'	=> "width:{$header_title['eael_data_table_header_col_img_size']}px;",
 										'alt'	=> esc_attr(get_post_meta($header_title['eael_data_table_header_col_img']['id'], '_wp_attachment_image_alt', true))
 									]);
-							?><img <?php echo $this->get_render_attribute_string('data_table_th_img'.$i); ?>><?php endif; ?><span class="data-table-header-text"><?php echo __( Helper::eael_wp_kses($header_title['eael_data_table_header_col']), 'essential-addons-for-elementor-lite'); ?></span></th>
+							?><img <?php $this->print_render_attribute_string('data_table_th_img'.$i); ?>><?php endif; ?><span class="data-table-header-text"><?php echo wp_kses( $header_title['eael_data_table_header_col'], Helper::eael_allowed_tags() ); ?></span></th>
```
Comment (robinwpdeveloper) [inline includes/Elements/Data_Table.php:1398]: `__()` translation support missing?

File: includes/Elements/Data_Table.php [hunk 2]
```diff
@@ -1311,13 +1308,11 @@ protected function render( ) {
 					'content_type'	=> $content_row['eael_data_table_content_type'],
 					'template'		=> $content_row['eael_primary_templates_for_tables'],
 	  				'title'			=> $tbody_content,
-	  				'link_url'		=> !empty($content_row['eael_data_table_content_row_title_link']['url'])?$content_row['eael_data_table_content_row_title_link']['url']:'',
+	  				'link_url'		=> ! empty($content_row['eael_data_table_content_row_title_link']) ? $content_row['eael_data_table_content_row_title_link'] : [],
 	  				'icon_content_new'	=> !empty($content_row['eael_data_table_icon_content_new']) ? $content_row['eael_data_table_icon_content_new']:'',
 	  				'icon_content'	=> !empty($content_row['eael_data_table_icon_content']) ? $content_row['eael_data_table_icon_content']:'',
 	  				'icon_migrated'	=> $icon_migrated,
 	  				'icon_is_new'	=> $icon_is_new,
-	  				'link_target'	=> $target,
```
Comment (robinwpdeveloper) [inline includes/Elements/Data_Table.php:1319]: Dont we need target and nofollow?

File: includes/Elements/Flip_Box.php [hunk 3]
```diff
@@ -1588,6 +1588,7 @@ protected function render()
 		                    if ( ! is_array( $settings['eael_flipbox_front_templates'] ) ) {
 			                    $settings['eael_flipbox_front_templates'] = apply_filters( 'wpml_object_id', $settings['eael_flipbox_front_templates'], 'wp_template', true );
 		                    }
+                            // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
```
Comment (robinwpdeveloper) [inline includes/Elements/Flip_Box.php:1591]: Looks like alignment issue with the next (below) line

File: includes/Elements/Flip_Box.php [hunk 4]
```diff
@@ -1597,17 +1598,23 @@ protected function render()
...
+                                    <?php if ( !empty( $settings['eael_flipbox_front_title'] ) ): 
+                                        $title = '<'. $settings['eael_flipbox_front_title_tag'] . ' class="eael-elements-flip-box-heading">';
+                                        $title .= $settings['eael_flipbox_front_title'];
+                                        $title .= '</' . $settings['eael_flipbox_front_title_tag'] . '>';
+                                        echo wp_kses( $title, Helper::eael_allowed_tags() );
+                                        endif; ?>
                                     <div class="eael-elements-flip-box-content">
-	                                    <?php $tagsPresent = preg_match( '/<(h[1-6]|p|pre)>.*<\/(h[1-6]|p|pre)>/i', $settings['eael_flipbox_front_text'] ); ?>
-                                        <?php echo $tagsPresent ? $settings['eael_flipbox_front_text'] : '<p>' . $settings['eael_flipbox_front_text'] . '</p>'; ?>
+	                                    <?php
```
Comment (robinwpdeveloper) [inline includes/Elements/Flip_Box.php:1615]: Makes sure there is no alignment issue.

File: includes/Elements/FluentForm.php [hunk 5]
```diff
@@ -77,14 +77,13 @@ public static function get_fluent_forms_list()
 
 		$options = array();
 
-		if (defined('FLUENTFORM')) {
-			global $wpdb;
+		if ( defined('FLUENTFORM') && function_exists( 'wpFluent' ) ) {
```
Comment (robinwpdeveloper) [inline includes/Elements/FluentForm.php:80]: These changes related to PCP?

**Review by robinwpdeveloper — COMMENTED** [review]
Left few queries.

File: includes/Elements/Data_Table.php (hunk 1 above)
Comment (jakaria-istauk) [reply includes/Elements/Data_Table.php:1398]: This is a dynamic value which is input by user from elementor editor. we didn't give translation support for any value input/changeable by user. Here may be mistakenly added. so i removed that.

File: includes/Elements/FluentForm.php (hunk 5 above)
Comment (jakaria-istauk) [reply includes/Elements/FluentForm.php:80]: yes. they suggest not to use direct $wpdb.

File: includes/Elements/Flip_Box.php (hunk 3 above)
Comment (jakaria-istauk) [reply includes/Elements/Flip_Box.php:1591]: In local editor it looks fine. 
image

**Review by robinwpdeveloper — APPROVED** [review]

**Pull request merged**